inc/tasks/promotions.php

```diff
@@ -203,7 +203,15 @@ function task_promotions($task)
 			}
 
 			$uids[] = $user['uid'];
+			if((int)$user['displaygroup'] === (int)$user['usergroup'] || in_array((int)$user['displaygroup'], array_map('intval', explode(',', $user['additionalgroups']))))
```

Comment:
From reading the code, it seems the conditional here should be inverted. It seems to me the code is resetting the display group of users even though they are members of such groups.

Maybe further testing is advised.